ledger-core/v2/conveyor/conveyor.go

```diff
@@ -118,16 +118,26 @@ func (p *PulseConveyor) GetPublishedGlobalAliasAndBargeIn(key interface{}) (smac
 }
 
 func (p *PulseConveyor) AddInput(ctx context.Context, pn pulse.Number, event InputEvent) error {
+	return p.AddInputExt(ctx, pn, event, false, smachine.CreateDefaultValues{})
+}
+
+func (p *PulseConveyor) AddInputExt(ctx context.Context, pn pulse.Number, event InputEvent, prevPulse bool,
+	createDefaults smachine.CreateDefaultValues,
+) error {
 	pulseSlotMachine, targetPN, pulseState, err := p.mapToPulseSlotMachine(pn)
 	switch {
 	case err != nil:
 		return err
 	case pulseSlotMachine == nil || pulseState == 0:
 		return fmt.Errorf("slotMachine is missing: pn=%v", pn)
 	}
+	if prevPulse {
+		pr, _ := pulseSlotMachine.pulseSlot.pulseData.PulseRange()
+		return p.AddInputExt(ctx, pr.LeftBoundNumber(), event, false, createDefaults)
+	}
+	// pulseSlotMachine.pulseSlot.prevPulse
```

Comment:
this comment makes no sense, should be deleted

Reply:
left accidentally. removed